server/configuration.go

```diff
@@ -80,5 +82,12 @@ func (p *Plugin) OnConfigurationChange() error {
 
 	p.setConfiguration(configuration)
 
+	if p.isActivated() {
```

Comment:
Why do you need to check this?
1. If the plugin get disabled, the hook will not be called.
2. `p.activated` gets set to `true` at startup.

Reply:
OnConfigurationChange actually gets called before OnActivate. I'm not sure if that's a bug or intended so that the plugin use its configuration in OnActivate (I also filed [a ticket for that](https://mattermost.atlassian.net/browse/MM-14788). I just added this check to be extra sure since I was worried about any potential weirdness there.

Reply:
Hm, I see. This feels like something the plugin system should support by default. But your solution is correct.

Reply:
Yeah, I can see reasons for doing it both ways. On one hand, I want to know if the configuration changes to set the plugin's config, even before OnActivate has been called. On the other hand, I assumed that it was only called after OnActivate, and I can see others making the same mistake